### PR TITLE
Feature/ju 313 java 21 migration fearghus

### PR DESCRIPTION
**Pull request by foman-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/JU-539

### Change description
Java 21 migration changes - upgrading dependencies to clear vulnerabilities, code cleanup, replacing deprecated features.

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by drobledomunoz-ch — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -201,7 +201,7 @@
                     </from>
                     <to>
                         //Once the pipeline is moved to shared services need to update this to image reference from shared services
-                        <image>169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/overseas-entities-api:latest</image>
+                        <image>169942020521.dkr.ecr.eu-west-2.amazonaws.com/local/overseas-entities-api:latest</image>
```
Comment (dgurjar-ch) [inline pom.xml:204]: I think the comment is not appropriate, It should be removed. 
Do we need local in image path

Comment (hwatkins-ch) [reply pom.xml:204]: Comment is still needed, this has not been updated to the shared services image reference yet

**Review by hwatkins-ch — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (foman-ch) [reply pom.xml:204]: thanks Dholoram, think we need this comment until it's moved to shared services - will add a note to the ticket to come back and remove

**Review by zrahman-ch — APPROVED** [review]
LGTM

**Review by aduffy-ch — APPROVED** [review]

**Pull request merged**